llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -5264,6 +5277,71 @@ SDValue ARMTargetLowering::LowerSELECT(SDValue Op, SelectionDAG &DAG) const {
                          SelectTrue, SelectFalse, ISD::SETNE);
 }
 
+SDValue BuildCtSelectMask(SDValue Cond, EVT MaskVT, SDLoc DL, SelectionDAG &DAG) {
+  Cond = DAG.getNode(ISD::AND, DL, MaskVT, Cond, DAG.getConstant(1, DL, MaskVT));
+  SDValue Zero = DAG.getConstant(0, DL, MaskVT);
+  return DAG.getNode(ISD::SUB, DL, MaskVT, Zero, Cond);  // mask = -cond
+}
+
+SDValue ARMTargetLowering::LowerCTSELECT(SDValue Op, SelectionDAG &DAG) const {
```

Comment:
It implements `CTSELECT` in custom lowering function. Should it be implemented using `CTSELECT` pseudo instruction and do expansion at later stage of pipeline?

Reply:
This should be completed now.